ops/main.py

```diff
@@ -167,33 +234,9 @@ def main(charm_class):
     debug = ('JUJU_DEBUG' in os.environ)
     setup_root_logging(model_backend, debug=debug)
 
-    # Process the Juju event relevant to the current hook execution
-    # JUJU_HOOK_NAME, JUJU_FUNCTION_NAME, and JUJU_ACTION_NAME are not used
-    # in order to support simulation of events from debugging sessions.
-    #
-    # TODO: For Windows, when symlinks are used, this is not a valid
-    #       method of getting an event name (see LP: #1854505).
-    juju_exec_path = Path(sys.argv[0])
-    has_dispatch = juju_exec_path.name == 'dispatch'
-    if has_dispatch:
-        # The executable was 'dispatch', which means the actual hook we want to
-        # run needs to be looked up in the JUJU_DISPATCH_PATH env var, where it
-        # should be a path relative to the charm directory (the directory that
-        # holds `dispatch`). If that path actually exists, we want to run that
-        # before continuing.
-        dispatch_path = juju_exec_path.parent / Path(os.environ['JUJU_DISPATCH_PATH'])
-        if dispatch_path.exists() and dispatch_path.resolve() != juju_exec_path.resolve():
-            argv = sys.argv.copy()
-            argv[0] = str(dispatch_path)
-            try:
-                subprocess.run(argv, check=True)
-            except subprocess.CalledProcessError as e:
-                logger.warning("hook %s exited with status %d", dispatch_path, e.returncode)
-                sys.exit(e.returncode)
-        juju_exec_path = dispatch_path
-    juju_event_name = juju_exec_path.name.replace('-', '_')
-    if juju_exec_path.parent.name == 'actions':
-        juju_event_name = '{}_action'.format(juju_event_name)
+    juju_event = _JujuEvent(charm_dir)
+    if juju_event.is_dispatch:
+        juju_event.run_any_legacy_hook()
```

Comment:
run_any_legacy_hook has the same check in it. Does it make sense to do both?

Reply:
No :-)